tycho-indexer/src/extractor/dynamic_contract_indexer.rs

```diff
@@ -0,0 +1,348 @@
+use std::{
+    collections::{HashMap, HashSet},
+    iter::zip,
+    sync::Arc,
+};
+
+use actix_web::web::trace;
+use tokio::sync::mpsc::{Receiver, Sender};
+use tracing::debug;
+use tycho_common::{
+    dto::{Block, Transaction},
+    models::{
+        blockchain::{EntryPoint, EntryPointWithData, TracedEntryPoint},
+        contract::AccountDelta,
+        Address, Chain, ChangeType, ComponentId, StoreKey, StoreVal,
+    },
+    storage::{EntryPointFilter, EntryPointGateway},
+    traits::{AccountExtractor, EntryPointTracer, StorageSnapshotRequest},
+    Bytes,
+};
+
+struct TransactionAccountUpdates {
+    tx: Transaction,
+    changes: Vec<ContractChanges>,
+}
+
+struct DCIBlockUpdate {
+    // A Hashmap mapping Transactions to the relevant AccountDeltas (either Create or Update)
+    // detected on the block.
+    result_deltas: HashMap<Transaction, Vec<AccountDelta>>,
+    // A Hashmap mapping component IDs to the new Accounts that were detected by the DCI on the
+    // block
+    result_components: HashMap<ComponentId, Vec<Address>>,
+}
```

Comment:
I wonder if we could just emit a regular block update here instead that is merged with the current BlockUpdate of the Extractor?